sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewCatalog.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connector.catalog;
+
+import java.util.Map;
+
+import org.apache.spark.annotation.DeveloperApi;
+import org.apache.spark.sql.catalyst.analysis.NoSuchNamespaceException;
+import org.apache.spark.sql.catalyst.analysis.NoSuchViewException;
+import org.apache.spark.sql.catalyst.analysis.ViewAlreadyExistsException;
+import org.apache.spark.sql.types.StructType;
+
+/**
+ * Catalog methods for working with views.
+ */
+@DeveloperApi
+public interface ViewCatalog extends CatalogPlugin {
```

Comment:
Why do we remove the following preserved properties from the [umbrella PR](https://github.com/apache/spark/pull/35636/files#diff-7094226eb616c14235b9a88fb6d8bc4bef39fa4f9879aa8dc4cfe4d031b720e7)?
```
  /**
   * A reserved property to specify the description of the view.
   */
  String PROP_COMMENT = "comment";

  /**
   * A reserved property to specify the owner of the view.
   */
  String PROP_OWNER = "owner";

  /**
   * A reserved property to specify the software version used to create the view.
   */
  String PROP_CREATE_ENGINE_VERSION = "create_engine_version";

  /**
   * A reserved property to specify the software version used to change the view.
   */
  String PROP_ENGINE_VERSION = "engine_version";

  /**
   * All reserved properties of the view.
   */
  List<String> RESERVED_PROPERTIES = Arrays.asList(
    PROP_COMMENT,
    PROP_OWNER,
    PROP_CREATE_ENGINE_VERSION,
    PROP_ENGINE_VERSION);
```

Reply:
I left them out as they felt more "implementation" then "API". Do they belong to ViewCatalog?

Reply:
I think they belong to `ViewCatalog` since there are similar properties in [TableCatalog](https://github.com/apache/spark/blob/5a71a7f7b5c1762677ddbfe39a7c35d645c25e94/sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableCatalog.java#L48)?

Reply:
They are similar these table properties. I had hoped to expose them with better APIs instead of string constants, but we can follow up possibly in Builder. So I will add them.